src/components/app/app.test.tsx

```diff
@@ -0,0 +1,102 @@
+import { fireEvent, render, screen, within } from '@testing-library/react'
+import '@testing-library/jest-dom'
+import { App } from './app'
+import React from 'react'
+import * as deck from '../../services/deck'
+import { CardSuit, CardValue } from '../../interfaces/card.interface'
+
+test('Renders app', () => {
+    render(<App />)
+
+    expect(screen.getByText('DEALER')).toBeInTheDocument
+    expect(screen.getByText('PLAYER')).toBeInTheDocument
+})
+
+test('Dealer is dealt a seven and ten', () => {
+    jest.spyOn(deck, 'dealHand').mockReturnValue({
+        cards: [
+            {
+                value: CardValue.SEVEN,
+                suit: CardSuit.SPADES,
+            },
+            {
+                value: CardValue.TEN,
+                suit: CardSuit.HEARTS,
+            },
+        ],
+        finished: true,
+    })
+
+    render(<App />)
+
+    fireEvent.click(screen.getByText('STAND'))
+
+    expect(
+        screen.getByText('DEALER').nextSibling.firstChild.nextSibling
+            .nextSibling.textContent
+    ).toBe('Total:  17')
+})
+
+test('Player is dealt two kings', () => {
+    jest.spyOn(deck, 'dealHand').mockReturnValue({
+        cards: [
+            {
+                value: CardValue.KING,
+                suit: CardSuit.SPADES,
+            },
+            {
+                value: CardValue.KING,
+                suit: CardSuit.HEARTS,
+            },
+        ],
+        finished: true,
+    })
+
+    render(<App />)
+    jest.resetAllMocks()
+    fireEvent.click(screen.getByText('SPLIT'))
+
+    expect(screen.getByText('PLAYER').nextSibling.childNodes.length).toBe(2)
+})
+
+test('Player won', () => {
+    let counter = 0
+    jest.spyOn(deck, 'dealHand').mockImplementation(() => {
+        if (counter == 0) {
```

Comment:
we could look into improving how the deck#dealHand is setup to make it easier to mock..

For now though jest has built in support for scenarios like this with its [mockReturnValueOnce](https://jestjs.io/docs/mock-functions#mock-return-values) function.

this code can be simplified to
```
    jest.spyOn(deck, 'dealHand').mockReturnValueOnce({
                cards: [
                    {
                        value: CardValue.SEVEN,
                        suit: CardSuit.SPADES,
                    },
                    {
                        value: CardValue.TEN,
                        suit: CardSuit.HEARTS,
                    },
                ],
                finished: true,
            }
    )
    .mockReturnValueOnce({
                cards: [
                    {
                        value: CardValue.KING,
                        suit: CardSuit.SPADES,
                    },
                    {
                        value: CardValue.KING,
                        suit: CardSuit.HEARTS,
                    },
                ],
                finished: true,
            }
    );
```


also let's integrate this with the idea above of helper function.. make a `mockDealHandOnce` in addition to `mockDealHand` that underneath calls `mockReturnValueOnce` and can be used in this test